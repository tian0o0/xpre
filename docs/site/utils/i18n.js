// component mixin
import { get } from '../../../src/utils';
import { camelize } from '../../../src/utils/format/string';

export default {
  computed: {
    $t() {
      const { name } = this.$options;
      const prefix = name ? camelize(name) + '.' : '';
      const messages = this.$vantMessages[this.$vantLang];

      return (path, ...args) => {
        const message = get(messages, prefix + path) || get(messages, path);
        return typeof message === 'function' ? message(...args) : message;
      };
    }
  }
};
