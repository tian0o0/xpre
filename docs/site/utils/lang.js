import Locale from '../../../src/components/locale';
import zhCN from '../../../src/components/locale/lang/zh-CN';
import enUS from '../../../src/components/locale/lang/en-US';

const langMap = {
  'en-US': {
    title: 'Xpre - Exploring Endless Possibilities',
    messages: enUS
  },
  'zh-CN': {
    title: 'Xpre | 一个差不多的组件库',
    messages: zhCN
  }
};

let currentLang = '';

function getDefaultLang() {
  const langs = Object.keys(langMap);
  const { hash } = location;

  for (let i = 0; i < langs.length; i++) {
    if (hash.indexOf(langs[i]) !== -1) {
      return langs[i];
    }
  }

  const userLang = localStorage.getItem('VANT_LANGUAGE') || navigator.language || 'en-US';
  return userLang.indexOf('zh-') !== -1 ? 'zh-CN' : 'en-US';
}

export function setLang(lang) {
  if (currentLang === lang) {
    return;
  }

  currentLang = lang;
  if (window.localStorage) {
    localStorage.setItem('VANT_LANGUAGE', lang);
  }
  Locale.use(lang, langMap[lang].messages);
  document.title = langMap[lang].title;
}

setLang(getDefaultLang());
