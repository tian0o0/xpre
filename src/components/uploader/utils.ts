export function toArray<T>(item: T | T[]): T[] {
  if (Array.isArray(item)) {
    return item;
  }

  return [item];
}

export function readFile(file: File, resultType: string) {
  return new Promise(resolve => {
    const reader = new FileReader();

    reader.onload = event => {
      resolve((event.target as FileReader).result);
    };

    if (resultType === 'dataUrl') {
      reader.readAsDataURL(file);
    } else if (resultType === 'text') {
      reader.readAsText(file);
    }
  });
}

export function isOversize(files: File | File[], maxSize: number): boolean {
  return toArray(files).some(file => file.size > maxSize);
}

export type FileListItem = {
  url?: string;
  file?: File;
  content?: string; // dataUrl
};

const IMAGE_EXT = ['jpeg', 'jpg', 'gif', 'png', 'svg'];

export function isImageUrl(url: string): boolean {
  return IMAGE_EXT.some(ext => url.indexOf(`.${ext}`) !== -1);
}

export function isImageFile(item: FileListItem): boolean {
  if (item.file && item.file.type) {
    return item.file.type.indexOf('image') === 0;
  }

  if (item.url) {
    return isImageUrl(item.url);
  }

  if (item.content) {
    return item.content.indexOf('data:image') === 0;
  }

  return false;
}
