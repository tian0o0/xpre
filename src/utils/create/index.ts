import { createBEM, Bem } from './bem';
import { createComponent, Component } from './component';
import { createI18N, Translate } from './i18n';

type CreateNamespaceReturn = [
  Component,
  Bem,
  Translate
];

export function createNamespace(name: string): CreateNamespaceReturn {
  name = 'x-' + name;
  return [createComponent(name), createBEM(name), createI18N(name)];
}
