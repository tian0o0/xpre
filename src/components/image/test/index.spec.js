
import { mount } from '../../../../test/utils';
import Image from '..';

test('click event', () => {
  const wrapper = mount(Image);

  wrapper.trigger('click');
  expect(wrapper.emitted('click')[0][0]).toBeTruthy();
  wrapper.destroy();
});

test('load event', () => {
  const wrapper = mount(Image, {
    propsData: {
      src: 'https://ae01.alicdn.com/kf/H6e5e22c9a0214cf692a69097c342e981T.png'
    }
  });

  wrapper.find('img').trigger('load');

  expect(wrapper.emitted('load')[0][0]).toBeTruthy();
  expect(wrapper).toMatchSnapshot();

  wrapper.setProps({ src: '' });
  expect(wrapper).toMatchSnapshot();
});

test('error event', () => {
  const wrapper = mount(Image, {
    propsData: {
      src: 'https://ae01.alicdn.com/kf/H6e5e22c9a0214cf692a69097c342e981T.png'
    }
  });

  wrapper.find('img').trigger('error');

  expect(wrapper.emitted('error')[0][0]).toBeTruthy();
});

test('lazy load', () => {
  const wrapper = mount(Image, {
    propsData: {
      src: 'https://ae01.alicdn.com/kf/H6e5e22c9a0214cf692a69097c342e981T.png',
      lazyLoad: true
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('lazy-load load event', done => {
  const wrapper = mount(Image, {
    propsData: {
      lazyLoad: true,
      src: 'https://ae01.alicdn.com/kf/H6e5e22c9a0214cf692a69097c342e981T.png'
    },
    mocks: {
      $Lazyload: {
        $on(eventName, hanlder) {
          if (eventName === 'loaded') {
            setTimeout(() => {
              hanlder({ el: null });
              hanlder({ el: wrapper.find('img').element });
              expect(wrapper.emitted('load').length).toEqual(1);
              expect(wrapper).toMatchSnapshot();
              wrapper.destroy();
            });
          }
        },
        $off() {
          done();
        }
      }
    }
  });
});

test('lazy-load error event', done => {
  const wrapper = mount(Image, {
    propsData: {
      lazyLoad: true
    },
    mocks: {
      $Lazyload: {
        $on(eventName, hanlder) {
          if (eventName === 'error') {
            setTimeout(() => {
              hanlder({ el: null });
              hanlder({ el: wrapper.find('img').element });
              expect(wrapper.emitted('error').length).toEqual(1);
              expect(wrapper).toMatchSnapshot();
              wrapper.destroy();
            });
          }
        },
        $off() {
          done();
        }
      }
    }
  });
});

test('show-loading prop', () => {
  const wrapper = mount(Image, {
    propsData: {
      showLoading: false
    }
  });

  expect(wrapper).toMatchSnapshot();
});

test('show-error prop', () => {
  const wrapper = mount(Image, {
    propsData: {
      showError: false,
      src: 'https://ae01.alicdn.com/kf/H6e5e22c9a0214cf692a69097c342e981T.png'
    }
  });

  wrapper.find('img').trigger('error');

  expect(wrapper).toMatchSnapshot();
});
