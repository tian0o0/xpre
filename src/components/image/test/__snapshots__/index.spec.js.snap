// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy load 1`] = `
<div class="x-image"><img class="x-image__img">
  <div class="x-image__loading"><i class="x-icon x-icon-photo-o" style="font-size: 22px;"></i></div>
</div>
`;

exports[`lazy-load error event 1`] = `
<div class="x-image">
  <div class="x-image__error"><i class="x-icon x-icon-warning-o" style="font-size: 22px;"></i></div>
</div>
`;

exports[`lazy-load load event 1`] = `<div class="x-image"><img class="x-image__img"></div>`;

exports[`load event 1`] = `<div class="x-image"><img src="https://img.yzcdn.cn/vant/cat.jpeg" class="x-image__img"></div>`;

exports[`load event 2`] = `
<div class="x-image"><img src="" class="x-image__img">
  <div class="x-image__loading"><i class="x-icon x-icon-photo-o" style="font-size: 22px;"></i></div>
</div>
`;

exports[`show-error prop 1`] = `<div class="x-image"></div>`;

exports[`show-loading prop 1`] = `<div class="x-image"><img class="x-image__img"></div>`;
